src/Plugin/Field/FieldFormatter/AvPortalVideoFormatter.php

```diff
@@ -263,10 +263,14 @@ protected function viewElement(FieldItemInterface $item): array {
       '#type' => 'html_tag',
       '#tag' => 'iframe',
       '#attributes' => [
+        'id' => $resource_ref,
```

Comment:
How do we prevent duplicated ID collisions?

Reply:
Same video 2x on the same page? I doubt that.

Reply:
As a developer, we should foresee these, we never know what a user could do... and god knows how creative a human can be! Isn't it ?
CC @ademarco

Reply:
I thinks we should use [Html::getUniqueId()](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Html.php/function/Html%3A%3AgetUniqueId/8.2.x) as suggested by @drupol , this would guarantee a valid HTML ID and its uniqueness.

Reply:
But that is not the requirement. This way we change the pattern for the ID completely.

Reply:
In the ticket, I read:

```
Currently, the AV Portal video iframe is missing some properties which are present both on the actual AV Portal website and the current Info site:

    'allowfullscreen' => TRUE,
    'webkitallowfullscreen' => TRUE,
    'mozallowfullscreen' => TRUE,
    'id' => 'videoplayerI167807' (where the ID is the ID of the video itself)

The scope of this ticket is to complete the iframe definition in the "AvPortalVideoFormatter"
```

And indeed, it says that the iframe should have an id which is the ID of the video. However, on the web, IDs needs to be checked to avoid duplicates or else we might end up with weird behaviour when Javascript is played (_and this will be probably the case here with the videoplayer_).

Maybe we should provide a new property like: `data-original-id="videoplayerI167807"` and let Drupal make sure that IDs are unique using the suggestion that @ademarco did previously.

What do you think?

Reply:
On top of that, using [Html::getUniqueId()](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Html.php/function/Html%3A%3AgetUniqueId/8.2.x) will not do any harm. It will only change the id if there are more than one in use, so we have to use it.

Reply:
Yeah we need to use `Html::getUniqueId()` so we don't end up generating invalid HTML if a user includes the same video twice.

Preventing the video from being shown twice is a lot harder I think.

Reply:
It looks like there is an uppercase character in the ID. This might cause issues since `Html::getUniqueId()` lowercases the IDs. It seems this is done because of HTML4 compatibility (according to the documentation of `\Drupal\Component\Utility\Html::getId()`) or because of Drupal HTML standards (according to the documentation of `\Drupal\Tests\Component\Utility\HtmlTest::providerTestHtmlGetUniqueId()`).

These IDs are case sensitive I believe so we will have to provide our own variant of `getUniqueId()` that preserves case.

Reply:
> These IDs are case sensitive I believe so we will have to provide our own variant of `getUniqueId()` that preserves case.

This is what I suggested indeed. Do you think we should do it in the scope of this ticket?

Reply:
Yes I think it can be done in scope of this ticket. The overhead does not need to be high since we can copy the code from `Html` and modify it. Some thoughts:

* Let's put a comment in there that links to https://www.drupal.org/project/drupal/issues/3034434 since when that issue is fixed we don't need our own version of this method any more.
* Let's put a check in that verifies if the passed in ID is already lowercased. If it is we should call the original implementation. Otherwise we cannot guarantee that the IDs are in fact unique, since there are two different static caches keeping track of the IDs.
   By checking if the passed in ID has any uppercased characters we can still guarantee uniqueness, since we will be keeping track of only uppercased ones, and core will track only lowercased ones.